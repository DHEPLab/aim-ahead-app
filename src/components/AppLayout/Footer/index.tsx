import React from "react";
import { Button } from "@mui/material";
import TipsAndUpdatesOutlinedIcon from "@mui/icons-material/TipsAndUpdatesOutlined";
import ZoomInOutlinedIcon from "@mui/icons-material/ZoomInOutlined";
import HelpOutlineOutlinedIcon from "@mui/icons-material/HelpOutlineOutlined";

import styles from "./index.module.scss";

const menus = [
  {
    Icon: TipsAndUpdatesOutlinedIcon,
    name: "BUILD SKILL",
  },
  {
    Icon: ZoomInOutlinedIcon,
    name: "DIAGNOSE",
  },
  {
    Icon: HelpOutlineOutlinedIcon,
    name: "SUBMIT A CASE",
  },
];

const Footer = () => {
  return (
    <div className={styles.footer}>
      {menus.map((menu) => {
        const { Icon, name } = menu;
        return (
          <Button key={name} className={styles.menu} sx={{ padding: "14px" }}>
            <div className={styles.icon}>
              <Icon sx={{ fontSize: "2rem" }} />
            </div>
            <span className={styles.name}>{name}</span>
          </Button>
        );
      })}
    </div>
  );
};

export default Footer;
