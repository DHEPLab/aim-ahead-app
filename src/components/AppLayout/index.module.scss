.mainLayout {
  display: flex;
  flex-direction: column;
}

.container {
  padding: 0 1rem;
  flex: 1;
}
