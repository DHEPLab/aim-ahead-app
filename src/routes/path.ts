export const path = {
  root: "/",
  signup: "/signup",
  login: "/login",
  resetPassword: "/reset-password/:token",
  case: "/case-review/:caseConfigId",
  answer: "/answer/:caseConfigId",
  forgotPassword: "/forgot-password",
};

export default path;
