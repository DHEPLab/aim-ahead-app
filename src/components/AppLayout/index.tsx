import React from "react";

import Header from "./Header";
import Footer from "./Footer";
import Layout from "../Layout";

import styles from "./index.module.scss";

const AppLayout = (props: React.PropsWithChildren) => {
  return (
    <Layout className={styles.mainLayout}>
      <Header />
      <div className={styles.container}>{props.children}</div>
      <Footer />
    </Layout>
  );
};

export default AppLayout;
