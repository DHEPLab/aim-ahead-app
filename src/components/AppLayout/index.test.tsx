import React, { ReactElement } from "react";
import { MemoryRouter, Route, Routes } from "react-router-dom";

import AppLayout from ".";
import path from "../../routes/path";
import { create } from "react-test-renderer";

const render = (element: ReactElement) => (
  <MemoryRouter initialEntries={[path.root]}>
    <Routes>
      <Route path={path.root} element={element} />
    </Routes>
  </MemoryRouter>
);

it("AppLayout match snapshot test", () => {
  expect(create(render(<AppLayout />)).toJSON()).toMatchSnapshot();
});
