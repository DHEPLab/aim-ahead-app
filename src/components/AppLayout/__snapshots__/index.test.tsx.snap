// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout match snapshot test 1`] = `
<div
  className="layout m mainLayout"
  data-testid="aim-ahead-layout"
  id="aim-ahead-layout"
>
  <div
    className="header"
  >
    <div>
      Collective Goods
    </div>
    <div
      className="right"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
      >
        <img
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="defalt_avatar.jpg"
        />
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DensityMediumIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 3h18v2H3zm0 16h18v2H3zm0-8h18v2H3z"
        />
      </svg>
    </div>
  </div>
  <div
    className="container"
  />
  <div
    className="footer"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary menu css-147avo0-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1q13jnw-MuiSvgIcon-root"
          data-testid="TipsAndUpdatesOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 20h4c0 1.1-.9 2-2 2s-2-.9-2-2m-2-1h8v-2H5zm11.5-9.5c0 3.82-2.66 5.86-3.77 6.5H5.27c-1.11-.64-3.77-2.68-3.77-6.5C1.5 5.36 4.86 2 9 2s7.5 3.36 7.5 7.5m-2 0C14.5 6.47 12.03 4 9 4S3.5 6.47 3.5 9.5c0 2.47 1.49 3.89 2.35 4.5h6.3c.86-.61 2.35-2.03 2.35-4.5m6.87-2.13L20 8l1.37.63L22 10l.63-1.37L24 8l-1.37-.63L22 6zM19 6l.94-2.06L22 3l-2.06-.94L19 0l-.94 2.06L16 3l2.06.94z"
          />
        </svg>
      </div>
      <span
        className="name"
      >
        BUILD SKILL
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary menu css-147avo0-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1q13jnw-MuiSvgIcon-root"
          data-testid="ZoomInOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14m.5-7H9v2H7v1h2v2h1v-2h2V9h-2z"
          />
        </svg>
      </div>
      <span
        className="name"
      >
        DIAGNOSE
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary menu css-147avo0-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1q13jnw-MuiSvgIcon-root"
          data-testid="HelpOutlineOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 18h2v-2h-2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8m0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4"
          />
        </svg>
      </div>
      <span
        className="name"
      >
        SUBMIT A CASE
      </span>
    </button>
  </div>
</div>
`;
